app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+import os
+
+
+def _check_path(new_path: str) -> None:
+    cur_path = ""
+    for directory in new_path.split("/"):
+        cur_path = os.path.join(cur_path, directory)
+        if not os.path.exists(cur_path):
+            os.mkdir(cur_path)
+
+
+def move_file(command: str) -> None:
+    try:
+        operation, old_file_loc, new_file_loc = command.split()
+        if not os.path.isfile(old_file_loc):
+            raise ValueError
+    except ValueError:
+        print(
+            "Please check the input, "
+            "it should be like 'mv file some_dir/new_file'. "
+            "The file to move cannot be directory!"
+        )
+        return
+    if operation == "mv" and old_file_loc != new_file_loc:
+        if "/" in new_file_loc:
+            new_path, *_ = new_file_loc.rsplit("/", 1)
+            if new_file_loc.endswith("/"):
+                if "/" in old_file_loc:
+                    _, old_file_name = old_file_loc.rsplit("/", 1)
+                    new_file_loc = new_path + "/" + old_file_name
+                else:
+                    new_file_loc = new_path + "/" + old_file_loc
+
+            _check_path(new_path)
```

Comment:
Will it works with the `\` instead of `/` ?) Can you try to simplify and make universally your code? You can use os.path.split(), os.makedirs() and other built-in functions to do this.

Reply:
great tip! tyvm